draft-ietf-quic-transport.md

```diff
@@ -1228,45 +1228,49 @@ version as part of the transport parameters (see {{transport-parameters}}).  As
 a result, modification of version negotiation packets by an attacker can be
 detected.
 
-The client includes two fields in the transport parameters:
-
-* The negotiated_version is the version that was finally selected for use.  This
-  MUST be identical to the value that is on the packet that carries the
-  ClientHello.  A server that receives a negotiated_version that does not match
-  the version of QUIC that is in use MUST terminate the connection with a
-  VERSION_NEGOTIATION_ERROR error code.
-
-* The initial_version is the version that the client initially attempted to use.
-  If the server did not send a version negotiation packet {{packet-version}},
-  this will be identical to the negotiated_version.
+The client includes the initial_version field in its transport parameters.  The
+initial_version is the version that the client initially attempted to use.  If
+the server did not send a version negotiation packet {{packet-version}}, this
+will be identical to the negotiated_version.
 
 A server that processes all packets in a stateful fashion can remember how
 version negotiation was performed and validate the initial_version value.
 
 A server that does not maintain state for every packet it receives (i.e., a
-stateless server) uses a different process. If the initial and negotiated
-versions are the same, a stateless server can accept the value.
-
-If the initial version is different from the negotiated_version, a stateless
-server MUST check that it would have sent a version negotiation packet if it had
-received a packet with the indicated initial_version.  If a server would have
-accepted the version included in the initial_version and the value differs from
-the value of negotiated_version, the server MUST terminate the connection with a
-VERSION_NEGOTIATION_ERROR error.
+stateless server) uses a different process. If the initial_version matches the
+version of QUIC that is in use, a stateless server can accept the value.
+
+If the initial_version is different from the version of QUIC that is in use, a
+stateless server MUST check that it would have sent a version negotiation packet
+if it had received a packet with the indicated initial_version.  If a server
+would have accepted the version included in the initial_version and the value
+differs from the QUIC version that is in use, the server MUST terminate the
+connection with a VERSION_NEGOTIATION_ERROR error.
+
+The server includes both the version of QUIC that is in use and a list of the
+QUIC versions that the server supports.
+
+The negotiated_version field is the version that is in use.  This MUST be set by
+the server to the value that is on the Initial packet that it accepts (not an
+Initial packet that triggers a Retry or Version Negotiation packet).  A client
```

Comment:
Why the Retry packet?

Reply:
Both are (potentially) stateless from the server perspective.  You *could* save the packet number in the Retry cookie, but I don't think we need to require that servers do that.  Note that the size of this cookie is strictly limited.  It has to fit into the padding space in the Initial packet.